api/types/provisioning.go

```diff
@@ -363,16 +367,31 @@ func (p *ProvisionTokenV2) Expiry() time.Time {
 	return p.Metadata.Expiry()
 }
 
-// GetName returns server name
+// GetName returns the name of the provision token. This value can be secret!
+// Use GetSafeName where the name may be logged.
 func (p *ProvisionTokenV2) GetName() string {
 	return p.Metadata.Name
 }
 
-// SetName sets the name of the TrustedCluster.
+// SetName sets the name of the provision token.
 func (p *ProvisionTokenV2) SetName(e string) {
 	p.Metadata.Name = e
 }
 
+// GetSafeName returns the name, sanitised appropriately for join methods
+// where the name is secret.
+func (p *ProvisionTokenV2) GetSafeName() string {
+	name := p.GetName()
+	if p.GetJoinMethod() != JoinMethodToken {
+		return name
+	}
+
+	hiddenBefore := int(0.75 * float64(len(name)))
```

Comment:
Is this censored enough ? I feel 75% will be fine for auto-generated tokens, but this could be too revealing on short static tokens configured by users. Perhaps below a certain length, we should just censor the whole thing here ?

Reply:
This is probably okay in general, but I agree it should maybe be dependent on length (and maybe use `math.Ceil()`)

If we're really concerned we could hash the token or something too (despite the inconvenience).